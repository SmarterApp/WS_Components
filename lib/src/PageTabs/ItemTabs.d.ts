import * as React from "react";
/**
 * Names of the different tab types.
 * @type {Tabs}
 */
export declare type Tabs = "viewer" | "rubric" | "information";
/**
 * ItemTabsProps props
 * @export
 * @interface ItemTabsProps
 */
export interface ItemTabsProps {
    changedTab: (tab: Tabs) => void;
    selectedTab: Tabs;
    showRubricTab?: boolean;
}
/**
 * ItemTabs displays options for view state of a selected item.
 * @export
 * @class ItemTabs
 * @extends {React.Component<ItemTabsProps, {}>}
 */
export declare class ItemTabs extends React.Component<ItemTabsProps, {}> {
    private select;
    constructor(props: ItemTabsProps);
    /**
     * event handler for view state.
     */
    onClick: (selectedVal: Tabs) => void;
    /**
     * renders individual tab components
     * @param {string} label
     * @param {Tabs} tabType
     * @returns
     */
    renderTab(label: string, tabType: Tabs): JSX.Element;
    render(): JSX.Element;
}
