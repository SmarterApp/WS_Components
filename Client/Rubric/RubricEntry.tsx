import * as React from 'react';
import * as AboutItemModels from '../AboutItem/AboutItemModels';
import * as Collapsible from './Collapsible';

export class RubricEntryComponent extends React.Component<AboutItemModels.RubricEntry, {}> {
    render() {
        const pointLabel = this.props.scorepoint === "1" ? "point" : "points";
        const label = `${this.props.name} (${this.props.scorepoint} ${pointLabel})`;
        return (
            <Collapsible.CComponent label={label}>
                <div dangerouslySetInnerHTML={{ __html: this.props.value }} />
            </Collapsible.CComponent>
        );
    }
}