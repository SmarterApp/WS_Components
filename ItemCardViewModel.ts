import * as GradeLevels from './GradeLevels'

export interface ItemCardViewModel {
    bankKey: number;
    itemKey: number;
    title: string;
    grade: GradeLevels.GradeLevels;
    gradeLabel: string;
    subjectCode: string;
    subjectLabel: string;
    claimCode: string;
    claimLabel: string;
    target: string;
    interactionTypeCode: string;
    interactionTypeLabel: string;
}