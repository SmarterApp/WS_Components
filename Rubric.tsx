import * as React from 'react';

export interface SampleResponse {
    purpose: string;
    scorePoint: string;
    name: string;
    sampleContent: string;
}

export interface RubricSample {
    maxValue: string;
    minValue: string;
    sampleResponses: SampleResponse[];
}

export interface RubricEntry {
    scorepoint: string;
    name: string;
    value: string;
}

export interface Rubric {
    language: string;
    rubricEntries: RubricEntry[];
    samples: RubricSample[];
}

export class RubricEntryComponent extends React.Component<RubricEntry, {}> {
    render() {
        const pointLabel = this.props.scorepoint === "1" ? "point" : "points";
        const label = `${this.props.name} (${this.props.scorepoint} ${pointLabel})`;
        return (
            <div dangerouslySetInnerHTML={{ __html: this.props.value }} />
        );
    }
}

export class SampleResponseComponent extends React.Component<SampleResponse, {}> {
    render() {
        const pointLabel = this.props.scorePoint == "1" ? "point" : "points";
        const label = `${this.props.name} (${this.props.scorePoint} ${pointLabel})`;
        return (
            <div className="sample-response">
                <b>Purpose: </b> {this.props.purpose} <br />
                <b>Sample Response: </b> <br />
                <div dangerouslySetInnerHTML={{ __html: this.props.sampleContent }} />
            </div>
        );
    }
}

export class RubricComponent extends React.Component<Rubric, {}> {

    renderRubrics() {
        const rubricEntries = this.props.rubricEntries.map((re, i) => <RubricEntryComponent {...re} key={String(i)} />);
        if (rubricEntries.length === 0) {
            return null;
        }

        let rubrics = (
            <div>
                <h4>Rubrics</h4>
                {rubricEntries}
            </div>
        );
        return rubrics;
    }

    renderSampleResponses() {
        let rubricSamples: JSX.Element[] = [];
        let i: number = 0;
        for (const sample of this.props.samples) {
            const key = `${i}:`;
            const responses = sample.sampleResponses.map((sr, idx) => <SampleResponseComponent {...sr} key={key + String(idx)} />);
            rubricSamples.push(...responses);
            i++;
        }
        if (rubricSamples.length === 0) {
            return null;
        }

        let sampleResponses = (
            <div>
                <h4>Sample Responses</h4>
                {rubricSamples}
            </div>
        );

        return sampleResponses;
    }

    render() {
        const label = `${this.props.language} Rubric`;
        return (
            <div className="item-information-content">
                {this.renderRubrics()}
                {this.renderSampleResponses()}
            </div>
        );
    }
}