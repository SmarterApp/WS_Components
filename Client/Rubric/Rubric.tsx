import * as React from 'react'
import * as AboutItemModels from '../AboutItem/AboutItemModels';
import * as RubricEntry from './RubricEntry';
import * as Collapsible from './Collapsible';
import * as SampleResponse from '../SampleResponse/SampleResponse';

export class RubricComponent extends React.Component<AboutItemModels.Rubric, {}> {

    renderRubrics() {
        const rubricEntries = this.props.rubricEntries.map((re, i) => <RubricEntry.RubricEntryComponent {...re} key={String(i)} />);
        if (rubricEntries.length === 0) {
            return null;
        }

        let rubrics = (
            <div>
                <h4>Rubrics</h4>
                {rubricEntries}
            </div>
        );
        return rubrics;
    }

    renderSampleResponses() {
        let rubricSamples: JSX.Element[] = [];
        let i: number = 0;
    
        for (const sample of this.props.samples) {
            const key = `${i}:`;
            const responses = sample.sampleResponses.map((sr, idx) => <SampleResponse.SampleResponseComponent {...sr} key={key + String(idx)} />);
            rubricSamples.push(...responses);
            i++;
        }

        if (rubricSamples.length === 0) {
            return null;
        }

        let sampleResponses = (
            <div>
                <h4>Sample Responses</h4>
                {rubricSamples}
            </div>
        );

        return sampleResponses;
    }

    render() {
        const label = `${this.props.language} Rubric`;
        return (
            <Collapsible.CComponent label={label}>
                {this.renderRubrics()}
                {this.renderSampleResponses()}
            </Collapsible.CComponent>
        );
    }
}