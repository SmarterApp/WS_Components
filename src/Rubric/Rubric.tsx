import * as React from "react";
import { Collapsible } from "./Collapsible";
import { RubricEntry } from "./RubricEntry";
import { RubricModel } from "./RubricModels";
import { SampleResponse } from "./SampleResponse";

export class Rubric extends React.Component<RubricModel, {}> {
  renderRubrics() {
    const rubricEntries = this.props.rubricEntries.map((re, i) => (
      <RubricEntry {...re} key={String(i)} />
    ));
    if (rubricEntries.length === 0) {
      return null;
    }

    const rubrics = (
      <div>
        <h4>Rubrics</h4>
        {rubricEntries}
      </div>
    );
    return rubrics;
  }

  renderSampleResponses() {
    const rubricSamples: JSX.Element[] = [];
    let i: number = 0;

    for (const sample of this.props.samples) {
      const key = `${i}:`;
      const responses = sample.sampleResponses.map((sr, idx) => (
        <SampleResponse {...sr} key={key + String(idx)} />
      ));
      rubricSamples.push(...responses);
      i++;
    }

    if (rubricSamples.length === 0) {
      return null;
    }

    const sampleResponses = (
      <div>
        <h4>Sample Responses</h4>
        {rubricSamples}
      </div>
    );

    return sampleResponses;
  }

  render() {
    const label = `${this.props.language} Rubric`;
    return (
      <Collapsible label={label}>
        {this.renderRubrics()}
        {this.renderSampleResponses()}
      </Collapsible>
    );
  }
}
