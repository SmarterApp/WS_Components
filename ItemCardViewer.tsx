import * as React from 'react';
import * as Rubric from './Rubric';
import * as AboutItem from './AboutItem';
import * as PageTabs from './PageTabs';
import * as ItemViewerFrame from './ItemViewerFrame';
import * as ItemInformation from './ItemInformation';
import * as ItemInformationDetail from './ItemInformationDetail';
import * as ApiModels from './ApiModels';

export interface Props {
    item?: { bankKey: number, itemKey: number }
}

export interface State {
    aboutItem: ApiModels.Resource<AboutItem.AboutThisItem>;
    selectedTab: PageTabs.Tabs;
}

export class ItemCardViewer extends React.Component<Props, State> {
    constructor(props: Props) {
        super(props);
        this.state = {
            selectedTab: "viewer",
            aboutItem: { kind: "loading" }
        }
    }

    getAboutItem() {
        if (this.props.item) {
            AboutItem.ScoreSearchClient(this.props.item)
                .then((data) => this.onSearchSuccess(data))
                .catch((err) => this.onSearchError(err));
        }
    }

    onSearchSuccess(data: AboutItem.AboutThisItem) {
        this.setState({
            aboutItem: { kind: "success", content: data }
        })
    }

    onSearchError(err: ExceptionInformation) {
        console.log(err);
    }

    onTabChange(tab: PageTabs.Tabs) {
        this.setState({
            selectedTab: tab
        });
    }

    renderViewer(url: string) {
        return (
            <div className="item-content">
                <ItemViewerFrame.ItemFrame url={url} />
            </div>
        );
    }

    renderRubric() {
        if (this.state.aboutItem.kind == "success" && this.state.aboutItem.content) {
            const rubrics = this.state.aboutItem.content.rubrics.map((ru, i) => <Rubric.RubricComponent {...ru } key={String(i)} />)
            return (
                <div className="item-content">{rubrics}</div>
            );
        }
    }

    renderInformation() {
        if (this.state.aboutItem.kind == "success" && this.state.aboutItem.content) {
            const aboutItem = this.state.aboutItem.content;
            return (
                <div className="item-content">
                    <div><ItemInformationDetail.ItemInformationDetail
                        itemCardViewModel={aboutItem.itemCardViewModel}
                        depthOfKnowledge={aboutItem.depthOfKnowledge}
                        commonCoreStandardsDescription={aboutItem.commonCoreStandardsDescription}
                        targetDescription={aboutItem.targetDescription}
                        educationalDifficulty={aboutItem.educationalDifficulty}
                        evidenceStatement={aboutItem.evidenceStatement} />
                    </div>
                </div>
            );
        }
    }

    renderChosen() {
        let selectedTab = null;
        if (this.state.aboutItem.kind == "success" && this.state.aboutItem.content) {
            const selectedTab = this.state.selectedTab;
            const itemCard = this.state.aboutItem.content.itemCardViewModel;

            let resultElement: JSX.Element[] | JSX.Element | undefined;
            if (selectedTab == "viewer") {
                const url = "http://ivs.smarterbalanced.org/items?ids=" + itemCard.bankKey.toString() + "-" + itemCard.itemKey.toString();
                resultElement = <div> {this.renderViewer(url)} </div>
            }
            else if (selectedTab == "rubric") {
                resultElement = <div> {this.renderRubric()} </div>
            }
            else if (selectedTab == "information") {
                resultElement = <div> {this.renderInformation()}</div>
            }
            return resultElement;
        }
    }

    render() {
        const tabs = PageTabs.ItemTabs;
        return (
            <div className="item-card">
                <PageTabs.ItemTabs changedTab={(tab) => this.onTabChange(tab)} selectedTab={this.state.selectedTab} />
                {this.renderChosen()}
            </div>
        );
    }
}