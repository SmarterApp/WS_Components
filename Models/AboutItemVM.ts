import * as Rubric from '../PageTabs/Rubric'
import * as ItemCardViewModel from './ItemCardViewModel'
import * as API from '../Models/ApiModels'

export const ScoreSearchClient = (params: { bankKey: number, itemKey: number }) => API.get<AboutThisItem>("/api/aboutItem", params)

export interface AboutThisItem {
    rubrics: Rubric.Rubric[];
    itemCardViewModel: ItemCardViewModel.ItemCardViewModel;
    depthOfKnowledge: string;
    targetDescription: string;
    commonCoreStandardsDescription: string;
    educationalDifficulty: string;
    evidenceStatement: string;
}