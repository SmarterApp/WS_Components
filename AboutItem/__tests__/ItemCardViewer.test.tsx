import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { ItemCardViewer } from '../ItemCardViewer';
import * as AboutItemVM from '../../Models/AboutItemVM';
import * as ItemCardViewModel from '../../Models/ItemCardViewModel'
import * as Rubric from '../../PageTabs/Rubric';
import * as TestUtils from 'react-dom/test-utils';
import { shallow, mount } from 'enzyme';

describe("ItemCardViewer", () => {
    const itemCardViewModel: ItemCardViewModel.ItemCardViewModel =  {
        bankKey: 1,
        itemKey: 123,
        title: "test_title",
        grade: 1,
        gradeLabel: "1",
        subjectCode: "Math",
        subjectLabel: "MTH",
        claimCode: "claim_1234",
        claimLabel: "claim_math",
        target: "",
        interactionTypeCode: "type01",
        interactionTypeLabel: "type01label",
        isPerformanceItem: false
    }

    const rubrics: Rubric.Rubric[] = []

    const item: AboutItemVM.AboutThisItem = {
        rubrics,
        itemCardViewModel,
        depthOfKnowledge: "depthOfKnowledge",
        targetDescription: "targetDescription",
        commonCoreStandardsDescription: "commonCoreStandardsDescription",
        educationalDifficulty: "educationalDifficulty",
        evidenceStatement: "evidenceStatement"
    }

    it("renders viewer tab", () => {
        let wrapper = shallow(<ItemCardViewer item={item}/>);
        wrapper.setState({selectedTab: "viewer"});
        expect(wrapper).toMatchSnapshot();
    })

    it("renders rubric tab", () =>  {
        let wrapper = shallow(<ItemCardViewer item={item}/>);
        wrapper.setState({selectedTab: "rubric"});
        expect(wrapper).toMatchSnapshot();
    })

    it("renders information tab", () =>  {
        let wrapper = shallow(<ItemCardViewer item={item}/>);
        wrapper.setState({selectedTab: "information"});
        expect(wrapper).toMatchSnapshot();
    })
})


