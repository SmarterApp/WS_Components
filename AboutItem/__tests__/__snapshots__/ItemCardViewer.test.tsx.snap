// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCardViewer renders information tab 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="item-card"
>
    <ItemTabs
        changedTab={[Function]}
        selectedTab="information"
    />
    <div>
         
        <div
            className="item-content"
        >
            <div>
                <ItemInformationDetail
                    commonCoreStandardsDescription="commonCoreStandardsDescription"
                    depthOfKnowledge="depthOfKnowledge"
                    educationalDifficulty="educationalDifficulty"
                    evidenceStatement="evidenceStatement"
                    itemCardViewModel={
                        Object {
                            "bankKey": 1,
                            "claimCode": "claim_1234",
                            "claimLabel": "claim_math",
                            "grade": 1,
                            "gradeLabel": "1",
                            "interactionTypeCode": "type01",
                            "interactionTypeLabel": "type01label",
                            "isPerformanceItem": false,
                            "itemKey": 123,
                            "subjectCode": "Math",
                            "subjectLabel": "MTH",
                            "target": "",
                            "title": "test_title",
                          }
                    }
                    targetDescription="targetDescription"
                />
            </div>
        </div>
    </div>
</div>,
  "nodes": Array [
    <div
      className="item-card"
>
      <ItemTabs
            changedTab={[Function]}
            selectedTab="information"
      />
      <div>
             
            <div
                  className="item-content"
            >
                  <div>
                        <ItemInformationDetail
                              commonCoreStandardsDescription="commonCoreStandardsDescription"
                              depthOfKnowledge="depthOfKnowledge"
                              educationalDifficulty="educationalDifficulty"
                              evidenceStatement="evidenceStatement"
                              itemCardViewModel={
                                    Object {
                                          "bankKey": 1,
                                          "claimCode": "claim_1234",
                                          "claimLabel": "claim_math",
                                          "grade": 1,
                                          "gradeLabel": "1",
                                          "interactionTypeCode": "type01",
                                          "interactionTypeLabel": "type01label",
                                          "isPerformanceItem": false,
                                          "itemKey": 123,
                                          "subjectCode": "Math",
                                          "subjectLabel": "MTH",
                                          "target": "",
                                          "title": "test_title",
                                        }
                              }
                              targetDescription="targetDescription"
                        />
                  </div>
            </div>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <ItemCardViewer
        item={
                Object {
                        "commonCoreStandardsDescription": "commonCoreStandardsDescription",
                        "depthOfKnowledge": "depthOfKnowledge",
                        "educationalDifficulty": "educationalDifficulty",
                        "evidenceStatement": "evidenceStatement",
                        "itemCardViewModel": Object {
                          "bankKey": 1,
                          "claimCode": "claim_1234",
                          "claimLabel": "claim_math",
                          "grade": 1,
                          "gradeLabel": "1",
                          "interactionTypeCode": "type01",
                          "interactionTypeLabel": "type01label",
                          "isPerformanceItem": false,
                          "itemKey": 123,
                          "subjectCode": "Math",
                          "subjectLabel": "MTH",
                          "target": "",
                          "title": "test_title",
                        },
                        "rubrics": Array [],
                        "targetDescription": "targetDescription",
                      }
        }
/>,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ItemCardViewer {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "item": Object {
            "commonCoreStandardsDescription": "commonCoreStandardsDescription",
            "depthOfKnowledge": "depthOfKnowledge",
            "educationalDifficulty": "educationalDifficulty",
            "evidenceStatement": "evidenceStatement",
            "itemCardViewModel": Object {
              "bankKey": 1,
              "claimCode": "claim_1234",
              "claimLabel": "claim_math",
              "grade": 1,
              "gradeLabel": "1",
              "interactionTypeCode": "type01",
              "interactionTypeLabel": "type01label",
              "isPerformanceItem": false,
              "itemKey": 123,
              "subjectCode": "Math",
              "subjectLabel": "MTH",
              "target": "",
              "title": "test_title",
            },
            "rubrics": Array [],
            "targetDescription": "targetDescription",
          },
        },
        "refs": Object {},
        "state": Object {
          "selectedTab": "information",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="item-card"
>
          <ItemTabs
                    changedTab={[Function]}
                    selectedTab="information"
          />
          <div>
                     
                    <div
                              className="item-content"
                    >
                              <div>
                                        <ItemInformationDetail
                                                  commonCoreStandardsDescription="commonCoreStandardsDescription"
                                                  depthOfKnowledge="depthOfKnowledge"
                                                  educationalDifficulty="educationalDifficulty"
                                                  evidenceStatement="evidenceStatement"
                                                  itemCardViewModel={
                                                            Object {
                                                                      "bankKey": 1,
                                                                      "claimCode": "claim_1234",
                                                                      "claimLabel": "claim_math",
                                                                      "grade": 1,
                                                                      "gradeLabel": "1",
                                                                      "interactionTypeCode": "type01",
                                                                      "interactionTypeLabel": "type01label",
                                                                      "isPerformanceItem": false,
                                                                      "itemKey": 123,
                                                                      "subjectCode": "Math",
                                                                      "subjectLabel": "MTH",
                                                                      "target": "",
                                                                      "title": "test_title",
                                                                    }
                                                  }
                                                  targetDescription="targetDescription"
                                        />
                              </div>
                    </div>
          </div>
</div>,
        "_debugID": 6,
        "_renderedOutput": <div
          className="item-card"
>
          <ItemTabs
                    changedTab={[Function]}
                    selectedTab="information"
          />
          <div>
                     
                    <div
                              className="item-content"
                    >
                              <div>
                                        <ItemInformationDetail
                                                  commonCoreStandardsDescription="commonCoreStandardsDescription"
                                                  depthOfKnowledge="depthOfKnowledge"
                                                  educationalDifficulty="educationalDifficulty"
                                                  evidenceStatement="evidenceStatement"
                                                  itemCardViewModel={
                                                            Object {
                                                                      "bankKey": 1,
                                                                      "claimCode": "claim_1234",
                                                                      "claimLabel": "claim_math",
                                                                      "grade": 1,
                                                                      "gradeLabel": "1",
                                                                      "interactionTypeCode": "type01",
                                                                      "interactionTypeLabel": "type01label",
                                                                      "isPerformanceItem": false,
                                                                      "itemKey": 123,
                                                                      "subjectCode": "Math",
                                                                      "subjectLabel": "MTH",
                                                                      "target": "",
                                                                      "title": "test_title",
                                                                    }
                                                  }
                                                  targetDescription="targetDescription"
                                        />
                              </div>
                    </div>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ItemCardViewer
    item={
        Object {
            "commonCoreStandardsDescription": "commonCoreStandardsDescription",
            "depthOfKnowledge": "depthOfKnowledge",
            "educationalDifficulty": "educationalDifficulty",
            "evidenceStatement": "evidenceStatement",
            "itemCardViewModel": Object {
              "bankKey": 1,
              "claimCode": "claim_1234",
              "claimLabel": "claim_math",
              "grade": 1,
              "gradeLabel": "1",
              "interactionTypeCode": "type01",
              "interactionTypeLabel": "type01label",
              "isPerformanceItem": false,
              "itemKey": 123,
              "subjectCode": "Math",
              "subjectLabel": "MTH",
              "target": "",
              "title": "test_title",
            },
            "rubrics": Array [],
            "targetDescription": "targetDescription",
          }
    }
/>,
}
`;

exports[`ItemCardViewer renders rubric tab 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="item-card"
>
    <ItemTabs
        changedTab={[Function]}
        selectedTab="rubric"
    />
    <div>
         
        <div
            className="item-content"
        >
            
        </div>
         
    </div>
</div>,
  "nodes": Array [
    <div
      className="item-card"
>
      <ItemTabs
            changedTab={[Function]}
            selectedTab="rubric"
      />
      <div>
             
            <div
                  className="item-content"
            >
                  
            </div>
             
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <ItemCardViewer
        item={
                Object {
                        "commonCoreStandardsDescription": "commonCoreStandardsDescription",
                        "depthOfKnowledge": "depthOfKnowledge",
                        "educationalDifficulty": "educationalDifficulty",
                        "evidenceStatement": "evidenceStatement",
                        "itemCardViewModel": Object {
                          "bankKey": 1,
                          "claimCode": "claim_1234",
                          "claimLabel": "claim_math",
                          "grade": 1,
                          "gradeLabel": "1",
                          "interactionTypeCode": "type01",
                          "interactionTypeLabel": "type01label",
                          "isPerformanceItem": false,
                          "itemKey": 123,
                          "subjectCode": "Math",
                          "subjectLabel": "MTH",
                          "target": "",
                          "title": "test_title",
                        },
                        "rubrics": Array [],
                        "targetDescription": "targetDescription",
                      }
        }
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ItemCardViewer {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "item": Object {
            "commonCoreStandardsDescription": "commonCoreStandardsDescription",
            "depthOfKnowledge": "depthOfKnowledge",
            "educationalDifficulty": "educationalDifficulty",
            "evidenceStatement": "evidenceStatement",
            "itemCardViewModel": Object {
              "bankKey": 1,
              "claimCode": "claim_1234",
              "claimLabel": "claim_math",
              "grade": 1,
              "gradeLabel": "1",
              "interactionTypeCode": "type01",
              "interactionTypeLabel": "type01label",
              "isPerformanceItem": false,
              "itemKey": 123,
              "subjectCode": "Math",
              "subjectLabel": "MTH",
              "target": "",
              "title": "test_title",
            },
            "rubrics": Array [],
            "targetDescription": "targetDescription",
          },
        },
        "refs": Object {},
        "state": Object {
          "selectedTab": "rubric",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="item-card"
>
          <ItemTabs
                    changedTab={[Function]}
                    selectedTab="rubric"
          />
          <div>
                     
                    <div
                              className="item-content"
                    >
                              
                    </div>
                     
          </div>
</div>,
        "_debugID": 4,
        "_renderedOutput": <div
          className="item-card"
>
          <ItemTabs
                    changedTab={[Function]}
                    selectedTab="rubric"
          />
          <div>
                     
                    <div
                              className="item-content"
                    >
                              
                    </div>
                     
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ItemCardViewer
    item={
        Object {
            "commonCoreStandardsDescription": "commonCoreStandardsDescription",
            "depthOfKnowledge": "depthOfKnowledge",
            "educationalDifficulty": "educationalDifficulty",
            "evidenceStatement": "evidenceStatement",
            "itemCardViewModel": Object {
              "bankKey": 1,
              "claimCode": "claim_1234",
              "claimLabel": "claim_math",
              "grade": 1,
              "gradeLabel": "1",
              "interactionTypeCode": "type01",
              "interactionTypeLabel": "type01label",
              "isPerformanceItem": false,
              "itemKey": 123,
              "subjectCode": "Math",
              "subjectLabel": "MTH",
              "target": "",
              "title": "test_title",
            },
            "rubrics": Array [],
            "targetDescription": "targetDescription",
          }
    }
/>,
}
`;

exports[`ItemCardViewer renders viewer tab 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="item-card"
>
    <ItemTabs
        changedTab={[Function]}
        selectedTab="viewer"
    />
    <div>
         
        <div
            className="item-content"
        >
            <ItemFrame
                url="http://ivs.smarterbalanced.org/items?ids=1-123"
            />
        </div>
         
    </div>
</div>,
  "nodes": Array [
    <div
      className="item-card"
>
      <ItemTabs
            changedTab={[Function]}
            selectedTab="viewer"
      />
      <div>
             
            <div
                  className="item-content"
            >
                  <ItemFrame
                        url="http://ivs.smarterbalanced.org/items?ids=1-123"
                  />
            </div>
             
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <ItemCardViewer
        item={
                Object {
                        "commonCoreStandardsDescription": "commonCoreStandardsDescription",
                        "depthOfKnowledge": "depthOfKnowledge",
                        "educationalDifficulty": "educationalDifficulty",
                        "evidenceStatement": "evidenceStatement",
                        "itemCardViewModel": Object {
                          "bankKey": 1,
                          "claimCode": "claim_1234",
                          "claimLabel": "claim_math",
                          "grade": 1,
                          "gradeLabel": "1",
                          "interactionTypeCode": "type01",
                          "interactionTypeLabel": "type01label",
                          "isPerformanceItem": false,
                          "itemKey": 123,
                          "subjectCode": "Math",
                          "subjectLabel": "MTH",
                          "target": "",
                          "title": "test_title",
                        },
                        "rubrics": Array [],
                        "targetDescription": "targetDescription",
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ItemCardViewer {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "item": Object {
            "commonCoreStandardsDescription": "commonCoreStandardsDescription",
            "depthOfKnowledge": "depthOfKnowledge",
            "educationalDifficulty": "educationalDifficulty",
            "evidenceStatement": "evidenceStatement",
            "itemCardViewModel": Object {
              "bankKey": 1,
              "claimCode": "claim_1234",
              "claimLabel": "claim_math",
              "grade": 1,
              "gradeLabel": "1",
              "interactionTypeCode": "type01",
              "interactionTypeLabel": "type01label",
              "isPerformanceItem": false,
              "itemKey": 123,
              "subjectCode": "Math",
              "subjectLabel": "MTH",
              "target": "",
              "title": "test_title",
            },
            "rubrics": Array [],
            "targetDescription": "targetDescription",
          },
        },
        "refs": Object {},
        "state": Object {
          "selectedTab": "viewer",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="item-card"
>
          <ItemTabs
                    changedTab={[Function]}
                    selectedTab="viewer"
          />
          <div>
                     
                    <div
                              className="item-content"
                    >
                              <ItemFrame
                                        url="http://ivs.smarterbalanced.org/items?ids=1-123"
                              />
                    </div>
                     
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="item-card"
>
          <ItemTabs
                    changedTab={[Function]}
                    selectedTab="viewer"
          />
          <div>
                     
                    <div
                              className="item-content"
                    >
                              <ItemFrame
                                        url="http://ivs.smarterbalanced.org/items?ids=1-123"
                              />
                    </div>
                     
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ItemCardViewer
    item={
        Object {
            "commonCoreStandardsDescription": "commonCoreStandardsDescription",
            "depthOfKnowledge": "depthOfKnowledge",
            "educationalDifficulty": "educationalDifficulty",
            "evidenceStatement": "evidenceStatement",
            "itemCardViewModel": Object {
              "bankKey": 1,
              "claimCode": "claim_1234",
              "claimLabel": "claim_math",
              "grade": 1,
              "gradeLabel": "1",
              "interactionTypeCode": "type01",
              "interactionTypeLabel": "type01label",
              "isPerformanceItem": false,
              "itemKey": 123,
              "subjectCode": "Math",
              "subjectLabel": "MTH",
              "target": "",
              "title": "test_title",
            },
            "rubrics": Array [],
            "targetDescription": "targetDescription",
          }
    }
/>,
}
`;
